src/metabase/models/serialization/base.clj

```diff
@@ -216,19 +217,32 @@
 (defmethod extract-query :default [model-name _]
   (raw-reducible-query model-name))
 
+(defn- ->zoned-date-time
+  "Given an Instant, LocalDateTime, or ZonedDateTime, convert it to a ZonedDateTime at UTC.
+  LocalDateTime is treated as already being in UTC."
+  [t]
+  (cond
+    (instance? LocalDateTime  t) (.atZone            ^LocalDateTime  t (ZoneId/of "UTC"))
```

Comment:
We have the `java-time` library which has Clojure wrappers for most if not all of the the `java.time.*` methods. We also have the `metabase.util.date-2` namespace that has a ton of stuff for working with `java.time.*` classes. I'm mentioning this mostly so you don't have to reinvent the wheel here. `t/zoned-date-time` or `u.date-2/add-zone-to-local` might be things you can use here, and the bonus is they have tons of tests around them. I don't see any unit tests around this function so by reusing some existing stuff you can increase the test coverage of the SerDes v2 code for free